tool/run_e2e_tests.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
```

Comment:
In general I am not a fan of tests that can't be ran with just `dart test`. Can we make this be a regular Dart test?

Reply:
We _could_, but it also kind of already is, which is why I'm hesitant.

I also like `dart test` :) which is why I like that the underlying binary being built here, `json_codable_test.dart`, is using `package:test`, it's pretty much an exact copy of the normal test in `package:json`.

If we turn `run_e2e_tests.sh` into equivalent Dart code that runs with `dart test`, then when it runs `json_codable_test.dart` there will be a `dart test` running a nested `package:test` test, and I think it'll be pretty hard to tell what's going on, because the outer and inner test case messages, failure messages, etc, will be hard to distinguish.

When you're iterating on one of these tests I think what you should be running is effectively `dart test` but via `_macro_tool`, which means you won't be running the shell script, but the one "test case" that's failing.

Soooo maybe this is a good setup? If we end up wanting any more complexity then it definitely makes sense to have it in dart and not sh. There's still at least one more test type left to add, golden error tests, so maybe take another look after that, what do you think?

Thanks.